### PR TITLE
5 - GET /api/articles endpoint

### PR DESCRIPTION
**Pull request by FabianFernandezM**
Following actions have been taken:
- Added GET /api/articles endpoint. 
- Created controller and model for this endpoint. 
- Created testing suite for this endpoint.
- Updated readme file to reflect this new endpoint.
- Updated package json file to include jest sorted.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -105,4 +105,37 @@ describe("/api/articles/:article_id", () => {
...
         });
     });
+})
+
+describe("/api/articles", () => {
+    test("GET 200: Should return an object with an array on its 'articles' key", () => {
+        return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({body}) => {
```
Comment (pippaamy) [inline __tests__/app.test.js:115]: this test isnt necessary here :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -105,4 +105,37 @@ describe("/api/articles/:article_id", () => {
...
+        })
+    })
+    test("GET 200: Should return an array of article objects with the correct keys, ordered in DESC order by 'created_at' value", () => {
+        return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({body}) => {
+            const {articles} = body
+            expect(articles.length).toBe(13)
+            expect(articles).toBeSortedBy("created_at", {descending: true})
```
Comment (pippaamy) [inline __tests__/app.test.js:128]: I would have your sorted by in a separate test to the results

File: __tests__/app.test.js [hunk 3]
```diff
@@ -105,4 +105,37 @@ describe("/api/articles/:article_id", () => {
...
+            articles.forEach(article => {
+                expect(typeof article.author).toBe("string")
+                expect(typeof article.title).toBe("string")
+                expect(typeof article.article_id).toBe("number")
+                expect(typeof article.topic).toBe("string")
+                expect(typeof article.created_at).toBe("string")
```
Comment (pippaamy) [inline __tests__/app.test.js:134]: this is lush!

File: controllers/topics_controllers.js [hunk 4]
```diff
@@ -26,4 +26,14 @@ async function getArticleById(req, res, next) {
     }
 }
 
-module.exports = {getTopics, getEndpoints, getArticleById}
\ No newline at end of file
+async function getArticles(req, res, next) {
+    try {
+        const articles = await fetchArticles()
+        res.status(200).send({articles})
```
Comment (pippaamy) [inline controllers/topics_controllers.js:32]: this is great!

**Review by pippaamy — COMMENTED** [review]
Great job, remember to update your endpoints.json as you go :)

File: models/topics_models.js [hunk 5]
```diff
@@ -7,6 +7,22 @@ function fetchTopics() {
     })
 }
 
+function fetchArticles() {
+    const sqlStr = `SELECT articles.author, articles.title, articles.article_id,
+    articles.topic, articles.created_at, articles.votes, articles.article_img_url, 
+    COUNT(comments.comment_id) AS comment_count
+    FROM articles
+    LEFT JOIN comments
```
Comment (pippaamy) [inline models/topics_models.js:15]: I would stick with async await or promises - we're doing both between the controller and model

Comment (FabianFernandezM) [reply models/topics_models.js:15]: Hey Pippa! Thanks for your feedback! I did attempt this but for some reason async/await in the models was failing a couple of test, so I decided to keep async await on controllers whilst keeping promises in the models.

**Pull request merged**